tests/test_abstract_repr.py

```diff
@@ -81,6 +77,64 @@
 )
 
 
+@pytest.mark.parametrize(
+    "layout",
+    [
+        RegisterLayout([[0, 0], [1, 1]]),
+        TriangularLatticeLayout(10, 10),
+        RegisterLayout([[10, 0], [1, 10]], slug="foo"),
+    ],
+)
+def test_layout(layout: RegisterLayout):
+    ser_layout_str = layout.to_abstract_repr()
+    ser_layout_obj = json.loads(ser_layout_str)
+    assert ser_layout_obj.get("slug", None) == layout.slug
+
+    re_layout = RegisterLayout.from_abstract_repr(ser_layout_str)
+    assert layout == re_layout
+
+    with pytest.raises(TypeError, match="must be given as a string"):
+        RegisterLayout.from_abstract_repr(ser_layout_obj)
+
+    # Check the validation catches invalid entries
+    with pytest.raises(
+        jsonschema.exceptions.ValidationError, match="is too long"
+    ):
+        ser_layout_obj["coordinates"].append([0, 0, 0])
+        RegisterLayout.from_abstract_repr(json.dumps(ser_layout_obj))
```

Comment:
I don't get where this error is coming from, but it seems to work 😅

Reply:
The coordinates are 3D and we only support 2D in the schema. I can add a comment to clarify

Reply:
Ahh, that comes from `jsonschema`